tools/c7n_azure/c7n_azure/container_host/host.py

```diff
@@ -0,0 +1,379 @@
+                     if b.properties.content_settings.content_md5 != self.blob_cache.get(b.name)]
+
+        # Get all YAML files on disk that are no longer in blob storage
+        cached_policy_files = [f for f in os.listdir(self.policy_cache)
+                               if f.lower().endswith('.yml')]
+
+        removed_files = [f for f in cached_policy_files if f not in [b.name for b in blobs]]
+
+        if not (removed_files or new_blobs):
+            return
+
+        # Update a copy so we don't interfere with
+        # iterations on other threads
+        policies_copy = self.policies.copy()
+
+        for f in removed_files:
+            path = os.path.join(self.policy_cache, f)
+            self.unload_policy_file(path, policies_copy)
+
+        # Get updated YML files
+        for blob in new_blobs:
+            policy_path = os.path.join(self.policy_cache, blob.name)
+            if os.path.exists(policy_path):
+                self.unload_policy_file(policy_path, policies_copy)
+
+            client.get_blob_to_path(container, blob.name, policy_path)
+            self.load_policy(policy_path, policies_copy)
+            self.blob_cache.update({blob.name: blob.properties.content_settings.content_md5})
+
+        # Assign our copy back over the original
+        self.policies = policies_copy
+
+        if self.require_event_update:
+            self.update_event_subscriptions()
+
+    def load_policy(self, path, policies):
+        """
+        Loads a YAML file and prompts scheduling updates
+        :param path: Path to YAML file on disk
+        :param policies: Dictionary of policies to update
+        """
+        with open(path, "r") as stream:
+            try:
+                policy_config = yaml.safe_load(stream)
+
+                new_policies = PolicyCollection.from_data(policy_config, self.options)
+
+                if new_policies:
+                    for p in new_policies:
+                        log.info("Loading Policy %s from %s" % (p.name, path))
+                        policies.update({p.name: {'policy': p}})
+
+                        # Update periodic and set event update flag
+                        self.update_periodic(p)
+                        if p.data.get('mode', {}).get('events'):
+                            self.require_event_update = True
+
+            except Exception as exc:
+                log.error('Invalid policy file %s %s' % (path, exc))
+
+    def unload_policy_file(self, path, policies):
+        """
+        Unload a policy file that has changed or been removed.
+        Take the copy from disk and pop all policies from dictionary
+        and update scheduled jobs and event registrations.
+        """
+        with open(path, "r") as stream:
+            try:
+                policy_config = yaml.safe_load(stream)
+            except yaml.YAMLError as exc:
+                log.warning('Failure loading cached policy for cleanup %s %s' % (path, exc))
```

Comment:
Do you need to do `os.unlink` here?

Reply:
In general the disk cache is meant to mirror what is in blob storage as we reference those when YML files change and we need to unload.  In the case where no valid policies were loaded out of the file the behavior is the same regardless I guess, but could change if we end up doing something smarter with how we sync/cache.